packages/ui/src/components/popover/popover.tsx

```diff
@@ -0,0 +1,170 @@
+      !this._controlElement.contains(e.target) &&
+      this.state.open
+    ) {
+      this._closePopover();
+    }
+  };
+
+  private _openPopover = () => {
+    const {onOpen} = this.props;
+    this.setState({open: true}, () => {
+      if (onOpen) {
+        onOpen();
+      }
+    });
+  };
+
+  private _closePopover = () => {
+    const {onClose} = this.props;
+    this.setState({open: false}, () => {
+      if (onClose) {
+        onClose();
+      }
+    });
+  };
+
+  private _handleEscButtonPressed = (e: KeyboardEvent) => {
+    // check if ESC or Enter pressed
+    if (this.state.open && (e.keyCode === 27 || e.keyCode === 13)) {
+      this._closePopover();
+    }
+  };
+  private _handleClick = () => {
+    if (this.props.triggerMode === PopoverTriggerMode.Click) {
+      if (this.state.open) {
+        this._closePopover();
+      } else {
+        this._openPopover();
+      }
+    }
+  };
+  private _handleMouseEnter = () => {
+    if (
+      this.props.triggerMode === PopoverTriggerMode.Hover &&
+      !this.state.open
+    ) {
+      this._openPopover();
+    }
+  };
+  private _handleMouseLeave = () => {
+    if (this.props.triggerMode === PopoverTriggerMode.Hover) {
+      this._closeTimeout = setTimeout(this._closePopover, CLOSE_ON_HOVER_DELAY);
+    }
+  };
+  private _handleHoverOnPopover = () => {
+    if (this.props.triggerMode === PopoverTriggerMode.Hover) {
+      if (this.state.open && this._closeTimeout) {
+        clearTimeout(this._closeTimeout);
+        this._closeTimeout = null;
+      } else {
+        this._closePopover();
+      }
+    }
+  };
+  render(props: PopoverProps): JSX.Element | null {
+    if (!props.content || !props.children) {
+      return null;
+    }
+    const verticalAlignment =
+      props.verticalPosition === PopoverVerticalPositions.Top
+        ? PopoverVerticalPositions.Top
+        : PopoverVerticalPositions.Bottom;
+    const horizontalAlignment =
+      props.horizontalPosition === PopoverHorizontalPositions.Left
+        ? PopoverHorizontalPositions.Left
+        : PopoverHorizontalPositions.Right;
+    return (
+      <div className={styles.popoverContainer}>
+        <div
+          className="popover-anchor-container"
+          onClick={this._handleClick}
+          onMouseEnter={this._handleMouseEnter}
+          onMouseLeave={this._handleMouseLeave}
+          ref={node => {
+            this._controlElement = node;
+          }}>
+          {props.children}
+        </div>
+        <div
+          aria-expanded={this.state.open ? 'true' : 'false'}
+          tabIndex={-1}
+          onMouseEnter={this._handleHoverOnPopover}
+          onMouseLeave={this._handleHoverOnPopover}
+          className={[
+            props.className,
+            styles.popoverComponent,
+            this.state.open ? styles.visible : '',
+            styles[verticalAlignment],
+            styles[horizontalAlignment],
+          ].join(' ')}>
+          {props.content}
+        </div>
```

Comment:
please add another component named `popover-content`. 
1. This component will hold the content of this div.
2. It should be added only when state is `open`. otherwise it should not appear on the dom. 
3. the logic of `componentDidMount` and unmount should be part of the `popover-content`

@semarche-kaltura let me know if you think we should stick with the implemented approach.